doc/design/proposal-strip-types.md

```diff
@@ -0,0 +1,99 @@
+# Strip Types Design
+
+## Problem
+
+TypeScript support has been for a few years at the top of the list of features that users want to see in Node.js.
+Discussion has been on going for a few years [#43818](https://github.com/nodejs/node/issues/43818), and the main blocker has been the lack of a clear on long term support.
+TypeScript does not follow semver, and the Node.js project has been reluctant to commit to supporting TypeScript in the long term, given that the lifespan of an LTS version is 3 years.
+
+## Proposal
+
+The proposal is to add a new flag to the Node.js CLI, `---experimental-strip-types`, which will replace inline TypeScript types with whitespace.
+No type checking is performed, and types are discarded.
+
+Example:
+
+```typescript
+interface Foo {
+    bar: string;
+}
+
+throw new Error("Whitespacing");
+
+```
+
+Becomes:
+
+```javascript
+
+
+
+
+throw new Error("Whitespacing");
+
+```
+
+>Note that the missing types are replaced with whitespace, so the line numbers are preserved, so that sourcemaps are not needed.
+
+By removing types completely, we can avoid the need to commit to supporting TypeScript in the long term, as the feature is not a full TypeScript implementation. We won't need to worry about breaking changes in TypeScript, as we are not implementing them.
+TypeScript features that cannot be replaced by whitespace, such as enums, will error.
+As in JavaScript files, file extensions are required in `import` statements and `import()` expressions.
+TypeScript features that depend on settings within `tsconfig.json`, such as paths or converting module formats, are unsupported.
+This will solve the long term compatibility issue, but might not be as complete as a full TypeScript implementation.
+
+### Goals
+
+#### Determining module system
+
+TypeScript supports multiple module systems, such as CommonJS, ES Modules.
+We want to avoid the need to parse `tsconfig.json` to determine the module system, as this would require a full TypeScript implementation.
+We want to be able to support `.mts` and `.cts` files, and both module systems in `.ts`.
+
+#### TypeScript only features
+
+TypeScript has features that cannot be replaced by whitespace.
+The most prominent unsupported features that require transformation are:
+
+* `Enum`
+* `experimentalDecorators`
+* `namespaces`
```

Comment:
```suggestion
* `namespace` and `module` declaration
```

There is a "legacy" (is it?) `module` declaration in TypeScript. See the note at the top of https://www.typescriptlang.org/docs/handbook/namespaces.html. It is proposed to be deprecated: https://github.com/microsoft/TypeScript/pull/58007.